### DISCUSSION TITLE
Undo/redo and magnets

### DISCUSSION THREAD

**Discussion by francoisschwarzentruber** [tableaunoir/tableaunoir#101] [Ideas]
There are two options.

A. (current implementation) Undo/redo is for the board itself. Magnets live in another world. Operations performed on magnets are not registered as operations by the undo/redo mechanisms. The advantage is that we consider magnets as sort of "floating clipboards".

B. Undo/redo could be for the board itself and for magnet operations (adding magnets, remove, moving). The advantage would be that we could... create animations <3.

What do you think?

**Comment by tekhnee** [comment]
Not having a strong opinion on this one.

Option B sounds simpler and more likely to be the expected behavior, so I'd probably opt for this one.

**Reply by francoisschwarzentruber** [reply]
Thank you for your answer. I also think it is the simpler for the user. :)

**Comment by tekhnee** [comment]
@francoisschwarzentruber "user" _über allen_ 😅

**Comment by francoisschwarzentruber** [comment]
Hello @tekhnee, 

I really want to thank you. You opt for option B two years ago. Because of that, Tableaunoir now can handle animations and you can make slides.

:)
